evm/src/fuzz/invariant/executor.rs

```diff
@@ -0,0 +1,275 @@
+                                failed_invariants
+                                    .borrow_mut()
+                                    .insert(invariant.name.clone(), Some(error.clone()));
+                            }
+
+                            break 'fuzz_run
+                        }
+                    }
+
+                    // Generates the next call from the run using the recently updated dictionary.
+                    inputs.extend(
+                        strat
+                            .new_tree(&mut branch_runner.borrow_mut())
+                            .map_err(|_| TestCaseError::Fail("Could not generate case".into()))?
+                            .current(),
+                    );
+                }
+
+                // Before each run, we must reset the database state.
+                executor.borrow_mut().backend_mut().db = clean_db.clone();
+
+                // We clear all the targeted contracts created during this run.
+                if !created_contracts.is_empty() {
+                    let mut writable_targeted = targeted_contracts.write();
+                    for addr in created_contracts.iter() {
+                        writable_targeted.remove(addr);
+                    }
+                }
+
+                fuzz_cases.borrow_mut().push(FuzzedCases::new(fuzz_runs));
+
+                Ok(())
+            })
+            .err()
+            .map(|test_error| InvariantFuzzError {
+                test_error,
+                // return_reason: return_reason.into_inner().expect("Reason must be set"),
+                return_reason: "".into(),
+                revert_reason: "".into(), /* revert_reason.into_inner().expect("Revert error
+                                           * string must be set"), */
+                addr: invariant_address,
+                func: Some(ethers::prelude::Bytes::default()),
+                inner_sequence: vec![],
+            });
+
+        // TODO: only saving one sequence case per invariant failure. Do we want more?
+        Ok(Some(InvariantFuzzTestResult {
+            invariants: failed_invariants.into_inner(),
+            cases: fuzz_cases.into_inner(),
+            reverts: reverts.get(),
+        }))
+    }
+
+    /// Prepares certain structures to execute the invariant tests:
+    /// * Fuzz dictionary
+    /// * Targeted contracts
+    /// * Invariant Strategy
+    fn prepare_fuzzing(
+        &mut self,
+        invariant_address: Address,
+        test_contract_abi: &Abi,
+        test_options: InvariantTestOptions,
+    ) -> eyre::Result<(EvmFuzzState, FuzzRunIdentifiedContracts, SBoxedStrategy<Vec<BasicTxDetails>>)>
+    {
+        // Finds out the chosen deployed contracts and/or senders.
+        let (targeted_senders, targeted_contracts) =
+            self.select_contracts_and_senders(invariant_address, test_contract_abi)?;
+
+        if targeted_contracts.is_empty() {
+            eyre::bail!("No contracts to fuzz!");
```

Comment:
nit, however I wonder if this should be a panic since ending up here would probably that we should have contracts? Otherwise we wouldn't even know we had an invariant test

```suggestion
            eyre::bail!("No contracts to fuzz");
```

Reply:
It might happen that the user hasn't deployed any contract on `setUp()`. Or, he made the mistake of filtering out every contract with `excludeContracts`.